src/ut/baseresolver_test.cpp

```diff
@@ -305,6 +316,51 @@ TEST_F(BaseResolverTest, ARecordMakeUpUsedGray)
   EXPECT_THAT(targets, Contains(gray_record));
 }
 
+/// Test that the lazy target selection iterator returns true when the target
+/// has been set, and false otherwise
+TEST_F(BaseResolverTest, ARecordIteratorNextReturnValue)
+{
+  AddrInfo record;
+  AddrInfo expected_record = ResolverTest::ip_to_addr_info("3.0.0.0");
+
+  add_white_records(1);
+  BaseAddrIterator* it = resolve_iter();
+
+  // The value of record should be set by the iterator.
+  EXPECT_TRUE(it->next(record));
+  EXPECT_EQ(record, expected_record);
+
+  // The value of record should be left unchanged by the iterator.
+  EXPECT_FALSE(it->next(record));
+  EXPECT_EQ(record, expected_record);
```

Comment:
Or maybe keep this the way it is and have a separate testcase that just tests when there are no targets to return.